src/effects/defs.h

```diff
@@ -40,6 +40,10 @@ constexpr int kNumEffectsPerUnit = 4;
 
 const QString kNoEffectString = QStringLiteral("---");
 
+const QString kMixerProfile = QStringLiteral("[Mixer Profile]");
+const QString kHighEqFrequency = QStringLiteral("HiEQFrequency");
+const QString kLowEqFrequency = QStringLiteral("LoEQFrequency");
```

Comment:
The current solution is perfectly OK, but has a performance implications. 

Putting a const QString into a header (internal namespace) will construct (malloc) the QString again in every cpp file that includes the header. 
The memory allocation "may" take place when the QString is used the first time. 

This is in our case when moved to ConfigObject 
https://github.com/mixxxdj/mixxx/blob/dfde28dbe44375706abcf8846e7fd099c1c66a87/src/preferences/configobject.h#L20

It would be possible to share the strings across the translation units by moving this to the global (external) namespace by adding this to a defs.cpp as 
`extern const QString kMixerProfile = QStringLiteral("[Mixer Profile]");`

and make this: 
`extern const QString kMixerProfile;`

This way the memory allocation is only done a single time and than shared in whole mixxx binary

Maybe we can sourround it with a defs namespace or such.

We have more of these shared const QString in Mixxx, so I am not sure if it worth to target the issue here?

Reply:
Okay, thanks for the info. So, like in library prefs.h/cpp (where we could enclose the defs in namespaces `mixxx`, `library` & `prefs` to have `const ConfigKey kSyncTrackMetadataConfigKey` instead of
`const ConfigKey mixxx::library::prefs::kSyncTrackMetadataConfigKey`)

Out of curiosity, how does this compare to `#define` like in [src/defs_urls.h](https://github.com/mixxxdj/mixxx/blob/dfde28dbe44375706abcf8846e7fd099c1c66a87/src/defs_urls.h#L24)?

I wouldn't do it for these 3 strings here, and I didn't find noteworthy occurences of more than one `const QString k*` in header files. File a reminder if you know of more places where this could be improved?

Reply:
I tried
defs.h
```
extern const QString kEmptyEffect;
```
new defs.ccp
```
const QString kEmptyEffect = QStringLiteral("---");
```
but get linker errors.

When omitting `extern` it builds and works well.

Reply:
@ronso0 My guess is it does not work, because defs.h is not included in defs.cpp. 

defs.h 
`extern const QString kEmptyEffect;` 
extern is here is a prototype definition in the global namespace it means 
* link to a kEmptyEffect from the global namespace. 
* Don't initialize or call constructor

defs.cpp 
`extern const QString kEmptyEffect;` via `#include "defs.h"`
`const QString kEmptyEffect = QStringLiteral("---");`
means instantiate and initialize kEmptyEffect. 
Use the publish it as kEmptyEffect in the global namespace. 

You may also join extern and the initialization 
`extern const QString kEmptyEffect = QStringLiteral("---");`

I think we should prefer that to indicate that it is extern visible. 

The allocation is done before main and leaked finally. 

Advantage: we have only a single allocation. 
Disadvantage cutter the global namespace (can be fixed by a named namespace).

Reply:
> I thought the entire point of the QStringLiteral macro was that there was no allocation taking place?!?

Oh right, my bad. We have no heap allocation here. 

The uinicode string along with 20 bytes of control data is used directly from the text segment and of every compilation unit that includes and uses the string. This is almost like a constexpr with reference counting disabled. 
An addition there is a pointer to that data in the bss initalized with that location.    

So there is no performance benefit, but a negligible memory impact. 

Sorry for he noise.

Reply:
Right, moreover, what about initialization order? The initialization order of TU's is undefined. If you declare something as extern const, you need to take care that no other global variable is being initialized based on it. This reeks of a potential source of bugs. And even if the string is `QStringLiteral`, the constructor for that variable still needs to run in order for the variable to be usable, even if the constructor doesn't do much, right?

Reply:
> If you declare something as extern const, you need to take care that no other global variable is being initialized based on it.

Yes. In this case it is no issue, because the constructor boils down to a pointer initializer. No
Other dependencys.

Reply:
> @ronso0 My guess is it does not work, because defs.h is not included in defs.cpp.

Aaahm, nope. I also tried namespace(s) to no avail.
The pattern you suggest works well with ConfigKey (like in library_prefs.h/cpp) but appearantly not with QString.
Btw I stopped investigating what's wrong.

Reply:
Yes, let's move on, not worth to discuss it in the scope of an unrelated PR.